cpp/betweenness_centrality_module/betweenness_centrality_online_module.cpp

```diff
@@ -0,0 +1,249 @@
+    const auto node_bc_scores = algorithm.Get(*graph, normalize);
+
+    for (const auto [node_id, bc_score] : node_bc_scores) {
+      InsertOnlineBCRecord(memgraph_graph, result, memory, node_id, bc_score);
+    }
+  } catch (const std::exception &e) {
+    mgp::result_set_error_msg(result, e.what());
+    return;
+  }
+}
+
+void Update(mgp_list *args, mgp_graph *memgraph_graph, mgp_result *result, mgp_memory *memory) {
+  try {
+    const auto created_nodes = mgp::value_get_list(mgp::list_at(args, 0));
+    const auto created_edges_list = mgp::value_get_list(mgp::list_at(args, 1));
+    const auto deleted_nodes = mgp::value_get_list(mgp::list_at(args, 2));
+    const auto deleted_edges_list = mgp::value_get_list(mgp::list_at(args, 3));
+    const auto normalize = mgp::value_get_bool(mgp::list_at(args, 4));
+    const auto threads = mgp::value_get_int(mgp::list_at(args, 5));
+
+    auto graph = mg_utility::GetGraphView(memgraph_graph, result, memory, mg_graph::GraphType::kUndirectedGraph);
+    std::unordered_map<uint64_t, double> node_bc_scores;
+
+    if (!::initialized) {
+      node_bc_scores = algorithm.Set(*graph, DEFAULT_DIRECTED, normalize, threads);
+      ::initialized = true;
+    } else {
+      const auto created_node_ids = mg_utility::GetNodeIDs(created_nodes);
+      const auto created_edges = mg_utility::GetEdgeEndpointIDs(created_edges_list);
+      const auto deleted_node_ids = mg_utility::GetNodeIDs(deleted_nodes);
+      const auto deleted_edges = mg_utility::GetEdgeEndpointIDs(deleted_edges_list);
+
+      if (created_node_ids.size() == 0 && deleted_node_ids.size() == 0) {  // Use the online algorithm
+        // Construct the graph as before the update
+        std::vector<std::pair<std::uint64_t, std::uint64_t>> edges;
+        edges.reserve(graph->Edges().size());
+        for (const auto edge : graph->Edges()) {
+          std::pair<uint64_t, uint64_t> edge_memgraph_ids = {graph->GetMemgraphNodeId(edge.from),
+                                                             graph->GetMemgraphNodeId(edge.to)};
+          // Newly created edges aren’t part of the prior graph
+          if (std::find(created_edges.begin(), created_edges.end(), edge_memgraph_ids) == created_edges.end())
+            edges.push_back({edge.from, edge.to});
+        }
+        for (const auto edge : deleted_edges) {
+          edges.push_back({graph->GetInnerNodeId(edge.first), graph->GetInnerNodeId(edge.second)});
+        }
+        auto prior_graph = mg_generate::BuildGraph(graph->Nodes().size(), edges, mg_graph::GraphType::kUndirectedGraph);
+
+        // Dynamically update betweenness centrality scores by each created edge
+        for (const auto created_edge : created_edges) {
+          edges.push_back({graph->GetInnerNodeId(created_edge.first), graph->GetInnerNodeId(created_edge.second)});
+          graph = mg_generate::BuildGraph(graph->Nodes().size(), edges, mg_graph::GraphType::kUndirectedGraph);
```

Comment:
This looks extremely computationally heavy

Reply:
The alternative is using the `CreateEdge` and `EraseEdge` methods. Whereas that will not mess up the inner IDs (with edge update, both the prior and the current graph have the same nodes), I'd appreciate a yes from you before going ahead with it.

Reply:
Since we've discussed this separately, please make a TODO for that somewhere :D